frontend/src/lib/components/ExportButton/exporterLogic.ts

```diff
@@ -64,57 +64,62 @@ export const exporterLogic = kea<exporterLogicType>([
 
     listeners(({ actions, props }) => ({
         exportItem: async ({ exportFormat, exportContext, successCallback }) => {
-            lemonToast.info(`Export started...`)
-
-            const trackingProperties = {
-                export_format: exportFormat,
-                dashboard: props.dashboardId,
-                insight: props.insightId,
-                total_time_ms: 0,
-            }
-            const startTime = performance.now()
-
-            try {
-                let exportedAsset = await api.exports.create({
+            const poller = new Promise(async (resolve, reject) => {
+                const trackingProperties = {
                     export_format: exportFormat,
                     dashboard: props.dashboardId,
                     insight: props.insightId,
-                    ...(exportContext || {}),
-                })
-
-                if (!exportedAsset.id) {
-                    throw new Error('Missing export_id from response')
+                    total_time_ms: 0,
                 }
+                const startTime = performance.now()
+
+                try {
+                    let exportedAsset = await api.exports.create({
+                        export_format: exportFormat,
+                        dashboard: props.dashboardId,
+                        insight: props.insightId,
+                        ...(exportContext || {}),
+                    })
+
+                    if (!exportedAsset.id) {
+                        reject('Missing export_id from response')
```

Comment:
A `return` here will [probably help](https://stackoverflow.com/questions/32536049/do-i-need-to-return-after-early-resolve-reject) ;).

Reply:
#til " rejecting or resolving won't stop the execution of the rest of the function." 🤯